### PR TITLE
Improve modpow test generation

### PR DESCRIPTION
**Pull request by Rigidity**
Moves the `modpow` and `%` test generation into a Rust script rather than Python. I intend to move the other tests to Rust later as well.

This also improves the test coverage by including negative bases and moduli. Exponents remain positive (since it fails otherwise) and moduli must be non-zero to prevent division by zero.

Also varies the length of the bytes, as well as including more tests than previously.

### CODE REVIEW

File: tools/src/bin/generate-modpow-tests.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        let mut exponent = BigInt::from_signed_bytes_be(&bytes(0, 32));
+        if exponent < BigInt::ZERO {
+            exponent = -exponent;
+        }
+
+        // Generate a random modulus, but ensure it's non-zero.
+        let mut modulus = BigInt::from_signed_bytes_be(&bytes(0, 32));
+        if modulus == BigInt::ZERO {
+            modulus += 1;
+        }
```
Comment (arvidn) [inline tools/src/bin/generate-modpow-tests.rs:37]: you could `+ 1` unconditionally, and preserve the distribution

File: tools/src/bin/generate-modpow-tests.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "; This file was generated by tools/src/bin/generate-modpow-tests.rs\n\n".to_string();
+
+    for _ in 0..100 {
+        let base = BigInt::from_signed_bytes_be(&bytes(0, 32));
+
+        // Generate a random exponent, but ensure it's positive.
+        let mut exponent = BigInt::from_signed_bytes_be(&bytes(0, 32));
+        if exponent < BigInt::ZERO {
+            exponent = -exponent;
+        }
```
Comment (arvidn) [inline tools/src/bin/generate-modpow-tests.rs:31]: isn't there a way to treat the bytes as unsigned to begin with?

File: tools/src/bin/generate-modpow-tests.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    fs::write("./op-tests/test-modpow.txt", tests).unwrap();
+}
+
+// Convert a `BigInt` to a CLVM atom.
+fn to_atom(num: BigInt) -> Vec<u8> {
+    let mut allocator = Allocator::new();
+    let ptr = allocator.new_number(num).unwrap();
+    let atom = allocator.atom(ptr);
```
Comment (arvidn) [inline tools/src/bin/generate-modpow-tests.rs:83]: if all you're interested in is the length of the atom, there's a shortcut, `atom_len()`

**Review by arvidn — COMMENTED** [review]
looks good, just a few minor comments

File: tools/src/bin/generate-modpow-tests.rs (hunk 1 above)
Comment (Rigidity) [reply tools/src/bin/generate-modpow-tests.rs:37]: Well technically this won't always work since it could be `-1`

File: tools/src/bin/generate-modpow-tests.rs (hunk 3 above)
Comment (Rigidity) [reply tools/src/bin/generate-modpow-tests.rs:83]: Thanks

File: tools/src/bin/generate-modpow-tests.rs (hunk 2 above)
Comment (Rigidity) [reply tools/src/bin/generate-modpow-tests.rs]: Changed it to `from_bytes_be(Sign::Plus, ...)`

**Review by arvidn — APPROVED** [review]
did you verify that the output is deterministic?

**Comment by Rigidity** [comment]
It is

File: op-tests/test-modpow.txt [hunk 4]
```diff
@@ -1,100 +1,100 @@
 ; This file was generated by tools/src/bin/generate-modpow-tests.rs
 
-modpow 9341829497586649593865894363733490755438507694201597 556606291647424564814177367572458554425074604463581598568935778451121529 4373238041963667894698452839741832172073 => 645485963992042551624174571461937584879 | 26775
```
Comment (arvidn) [inline op-tests/test-modpow.txt]: how come this changed, if the generation is deterministic?

**Review by arvidn — APPROVED** [review]

File: op-tests/test-modpow.txt (hunk 4 above)
Comment (arvidn) [reply op-tests/test-modpow.txt]: right, you changed how the bytes are interpreted

Comment (Rigidity) [reply op-tests/test-modpow.txt]: I also switched out the RNG
